### PR TITLE
Create Demo Docker Image #1017

### PR DESCRIPTION
**Pull request by lauriejefferson**
I added the docker images for the rails app and the postgres database with the demo data loaded.  Both files are .tar.gz files.

### CODE REVIEW

File: db/schema.rb [hunk 1]
```diff
@@ -10,9 +10,12 @@
 #
 # It's strongly recommended that you check this file into your version control system.
 
-ActiveRecord::Schema.define(version: 2023_05_26_212613) do
+ActiveRecord::Schema.define(version: 2020_08_06_004459) do
```
Comment (johnbumgardner) [inline db/schema.rb:13]: hey i see you removed the conversion to active storage migration.

this right now is not working on postgres. could you keep the migrations the same and try rewriting the docker stuff to work with mysql or sqlite. i know the migrations work there.

File: docker-compose.yml [hunk 2]
```diff
@@ -5,23 +5,33 @@ services:
...
     environment:
-      RAILS_ENV: 'production'
-      PASSENGER_APP_ENV: 'production'
-      DB_DATABASE: 'fat_free_crm_production'
+      RAILS_ENV: 'development'
+      PASSENGER_APP_ENV: 'development'
+      DB_DATABASE: 'fat_free_crm_development'
       DB_USERNAME: 'postgres'
-      DB_PASSWORD: ''
+      DB_PASSWORD: 'password'
```
Comment (johnbumgardner) [inline docker-compose.yml:17]:
```suggestion
      DB_PASSWORD: ''
```

File: docker-compose.yml [hunk 3]
```diff
@@ -5,23 +5,33 @@ services:
...
-      - "80:80"
+      - '3000:3000'
+      - '80:80'
     volumes:
       - gems:/usr/local/rvm/gems
     environment:
-      RAILS_ENV: 'production'
-      PASSENGER_APP_ENV: 'production'
-      DB_DATABASE: 'fat_free_crm_production'
+      RAILS_ENV: 'development'
```
Comment (johnbumgardner) [inline docker-compose.yml:13]: if we're going to have this setup for development, can you rename this file to docker-compose-dev.yml

you can keep the production docker-compose file too

File: docker-compose.yml [hunk 4]
```diff
@@ -5,23 +5,33 @@ services:
...
-    image: postgres:9.5
+    image: postgres:12.15-alpine
+    container_name: db
     restart: always
     volumes:
       - pgdata:/var/lib/postgresql/data
+    environment:
+      POSTGRES_DB: fat_free_crm_development
+      POSTGRES_USER: postgres
+      POSTGRES_PASSWORD: password
```
Comment (johnbumgardner) [inline docker-compose.yml:29]:
```suggestion
      POSTGRES_PASSWORD: 
```

File: config/database.postgres.yml [hunk 5]
```diff
@@ -2,10 +2,10 @@
 development: &development
   adapter: postgresql
   database: fat_free_crm_development
-  username:
-  password:
-  host:     localhost
-  port:     5432
+  username: postgres
+  password: password
```
Comment (johnbumgardner) [inline config/database.postgres.yml:6]:
```suggestion
  password: 
```

File: Dockerfile [hunk 6]
```diff
@@ -21,10 +21,11 @@ RUN apt-get update && \
 	cp config/database.postgres.docker.yml config/database.yml && \
 	gem install bundler && \
 	bundle config set --local deployment 'true' && \
-	bundle install --deployment && \
-	bundle exec rails assets:precompile
+	bundle install --deployment 
+# bundle exec rails assets:precompile
```
Comment (johnbumgardner) [inline Dockerfile:25]: curious as to why this isnt needed. should improve performance for users

File: db/schema.rb (hunk 1 above)
Comment (lauriejefferson) [reply db/schema.rb:13]: I think I ran a migration and then rolled it back just trying to run the app to see how it works.  I wen back into the git branch for this pull request and it show 'ActiveRecord::Schema.define(version: 2020_08_06_004459)'.  Do you want images for both production and development for all three databases?

Comment (johnbumgardner) [reply db/schema.rb:13]: it's just for demo. just mysql should be sufficient.

**Comment by lauriejefferson** [comment]
I'm still working on this issue.  Just added some docker-compose files.

**Comment by lauriejefferson** [comment]
I made the above changes to the docker-compose files.  I added a new docker-compose file for MySQL.  The Rails docker container can't connect to the MySQL docker container through the socket.  I tried both MySQL:5.7.42 and MySQL:latest (8.33) and received the error "Can't connect to local MySQL server through socket'.  I checked to make sure that the socket was correctly configured in the Rails database.yml file and each time, with both MySQl versions, I received the same error. 
 Are there any specific configurations for MySQL that need to be setup to connect to the docker container?

**Comment by lauriejefferson** [comment]
The link to the mysql demo database is [here](https://www.dropbox.com/s/6xi8001tawlmv7o/fat_free_crm_mysql_bkup.tar.gz?dl=0). The file size exceeds GitHub's LFS Storage maximum, so I could not upload it in a git push.

**Comment by johnbumgardner** [comment]
can you add back the active storage stuff?

**Comment by lauriejefferson** [comment]
@johnbumgardner Can this PR be closed?

**Comment by lauriejefferson** [comment]
@johnbumgardner @brynary Do you still need these images?  Can this PR be closed?

**Comment by lauriejefferson** [comment]
@johnbumgardner @brynary @olleolleolle @endersonmaia Is this PR still needed?  Can it be closed?

**Pull request closed without merging**